storage.go

```diff
@@ -164,3 +237,38 @@ func (s *Storage) write(ctx context.Context, path string, r io.Reader, size int6
 
 	return size, nil
 }
+
+func (s *Storage) writeAppend(ctx context.Context, o *Object, r io.Reader, size int64, opt pairStorageWriteAppend) (n int64, err error) {
+	if !o.Mode.IsAppend() {
+		err = fmt.Errorf("object not appendable")
+		return
+	}
+
+	sessionId := GetObjectMetadata(o).UploadSessionID
+
+	offset, ok := o.GetAppendOffset()
+	if !ok {
+		err = fmt.Errorf("append offset is not set")
+		return
+	}
+
+	cursor := &files.UploadSessionCursor{
+		SessionId: sessionId,
+		Offset:    uint64(offset),
+	}
+
+	appendArg := &files.UploadSessionAppendArg{
+		Cursor: cursor,
+		Close:  false,
+	}
+
+	err = s.client.UploadSessionAppendV2(appendArg, r)
+	if err != nil {
+		return
+	}
+
+	offset += size
+	o.SetAppendOffset(offset)
+
+	return offset, nil
```

Comment:
`n` should be data that written instead of offset.

Reply:
Yes，I misunderstood the return value.